### ISSUE TITLE
unparsable section with reference to ephemeral dbt model (with same-name CTE?)

### ISSUE THREAD

**Issue by jdub** [sqlfluff/sqlfluff#1571]
I _suspect_ this has something to do with the CTE having the same name in the parent model and the ephemeral model, because if I rename `account_data` to `pants` in the parent model, the failure disappears.

## Expected Behaviour

No parsing error. 😆

## Observed Behaviour

```
$ sqlfluff lint
=== [dbt templater] Sorting Nodes...
=== [dbt templater] Compiling dbt project...
=== [dbt templater] Project Compiled.
== [/Users/jeffwaugh/src/dbt-sqlfluff/models/example/stg_accounts__accounts.sql] FAIL
L:   8 | P:  10 |  PRS | Line 13, Position 1: Found unparsable section:
                       | '),account_data as (\n\n select\n ...'
All Finished 📜 🎉!
```

## Steps to Reproduce

Here's a parent model that references an ephemeral model, `models/example/stg_accounts__accounts.sql`:

```sql
with

account_data as (

    select user_uuid, email from {{ ref('base_accounts__account_data') }}

)

select * from account_data
```

And here's the ephemeral model, doesn't matter if it's configured inline or in the project file, `models/example/base/base_accounts__account_data.sql`:

```sql
{{ config(materialized='ephemeral') }}

with

account_data as (

    select user_uuid, email from {{ source('accounts', 'account_data') }}

)

select * from account_data
```

## Version

```
$ python --version
Python 3.9.7

$ dbt --version
installed version: 0.21.0
   latest version: 0.21.0

Up to date!

Plugins:
  - bigquery: 0.21.0

$ sqlfluff --version
sqlfluff, version 0.6.9
```

## Configuration

```
[sqlfluff]
dialect = bigquery
templater = dbt

# https://docs.sqlfluff.com/en/stable/rules.html
# L016: Line is too long
# L034: Use wildcards then simple targets before calculations and aggregates in select statements
exclude_rules = L016,L034

[sqlfluff:rules:L014]
extended_capitalisation_policy = lower
```

## Parse

```
$ sqlfluff parse .
=== [dbt templater] Compiling dbt project...
=== [dbt templater] Project Compiled.
[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    newline:                                                  '\n'
[L:  2, P:  1]      |    newline:                                                  '\n'
[L:  3, P:  1]      |    statement:
[L:  3, P:  1]      |        with_compound_statement:
[L:  3, P:  1]      |            keyword:                                          'with'
[L:  3, P:  5]      |            newline:                                          '\n'
[L:  4, P:  1]      |            newline:                                          '\n'
[L:  5, P:  1]      |            common_table_expression:
[L:  5, P:  1]      |                identifier:                                   'account_data'
[L:  5, P: 13]      |                whitespace:                                   ' '
[L:  5, P: 14]      |                keyword:                                      'as'
[L:  5, P: 16]      |                whitespace:                                   ' '
[L:  5, P: 17]      |                bracketed:
[L:  5, P: 17]      |                    start_bracket:                            '('
[L:  5, P: 18]      |                    [META] indent:
[L:  5, P: 18]      |                    newline:                                  '\n'
[L:  6, P:  1]      |                    newline:                                  '\n'
[L:  7, P:  1]      |                    whitespace:                               '    '
[L:  7, P:  5]      |                    select_statement:
[L:  7, P:  5]      |                        select_clause:
[L:  7, P:  5]      |                            keyword:                          'select'
[L:  7, P: 11]      |                            [META] indent:
[L:  7, P: 11]      |                            newline:                          '\n'
[L:  8, P:  1]      |                            whitespace:                       '        '
[L:  8, P:  9]      |                            select_clause_element:
[L:  8, P:  9]      |                                column_reference:
[L:  8, P:  9]      |                                    identifier:               'user_uuid'
[L:  8, P: 18]      |                            comma:                            ','
[L:  8, P: 19]      |                            newline:                          '\n'
[L:  9, P:  1]      |                            whitespace:                       '        '
[L:  9, P:  9]      |                            select_clause_element:
[L:  9, P:  9]      |                                column_reference:
[L:  9, P:  9]      |                                    identifier:               'email'
[L:  9, P: 14]      |                        newline:                              '\n'
[L: 10, P:  1]      |                        whitespace:                           '    '
[L: 10, P:  5]      |                        [META] dedent:
[L: 10, P:  5]      |                        from_clause:
[L: 10, P:  5]      |                            keyword:                          'from'
[L: 10, P:  9]      |                            whitespace:                       ' '
[L: 10, P: 10]      |                            from_expression:
[L: 10, P: 10]      |                                [META] indent:
[L: 10, P: 10]      |                                from_expression_element:
[L: 10, P: 10]      |                                    table_expression:
[L: 10, P: 10]      |                                        table_reference:
[L: 10, P: 10]      |                                            identifier:       '`dataset`'
[L: 10, P: 10]      |                                            dot:              '.'
[L: 10, P: 10]      |                                            identifier:       '`accounts_public`'
[L: 10, P: 10]      |                                            dot:              '.'
[L: 10, P: 10]      |                                            identifier:       '`account_data_*`'
[L: 10, P: 50]      |                                [META] dedent:
[L: 10, P: 50]      |                    newline:                                  '\n'
[L: 11, P:  1]      |                    newline:                                  '\n'
[L: 12, P:  1]      |                    [META] dedent:
[L: 12, P:  1]      |                    end_bracket:                              ')'
[L: 12, P:  2]      |            newline:                                          '\n'
[L: 13, P:  1]      |            newline:                                          '\n'
[L: 14, P:  1]      |            select_statement:
[L: 14, P:  1]      |                select_clause:
[L: 14, P:  1]      |                    keyword:                                  'select'
[L: 14, P:  7]      |                    [META] indent:
[L: 14, P:  7]      |                    whitespace:                               ' '
[L: 14, P:  8]      |                    select_clause_element:
[L: 14, P:  8]      |                        wildcard_expression:
[L: 14, P:  8]      |                            wildcard_identifier:
[L: 14, P:  8]      |                                star:                         '*'
[L: 14, P:  9]      |                whitespace:                                   ' '
[L: 14, P: 10]      |                [META] dedent:
[L: 14, P: 10]      |                from_clause:
[L: 14, P: 10]      |                    keyword:                                  'from'
[L: 14, P: 14]      |                    whitespace:                               ' '
[L: 14, P: 15]      |                    from_expression:
[L: 14, P: 15]      |                        [META] indent:
[L: 14, P: 15]      |                        from_expression_element:
[L: 14, P: 15]      |                            table_expression:
[L: 14, P: 15]      |                                table_reference:
[L: 14, P: 15]      |                                    identifier:               'account_data'
[L: 14, P: 27]      |                        [META] dedent:
[L: 14, P: 27]      |    newline:                                                  '\n'

[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    statement:
[L:  1, P:  1]      |        with_compound_statement:
[L:  1, P:  1]      |            keyword:                                          'with'
[L:  1, P:  5]      |            newline:                                          '\n'
[L:  2, P:  1]      |            newline:                                          '\n'
[L:  3, P:  1]      |            common_table_expression:
[L:  3, P:  1]      |                identifier:                                   'account_data'
[L:  3, P: 13]      |                whitespace:                                   ' '
[L:  3, P: 14]      |                keyword:                                      'as'
[L:  3, P: 16]      |                whitespace:                                   ' '
[L:  3, P: 17]      |                bracketed:
[L:  3, P: 17]      |                    start_bracket:                            '('
[L:  3, P: 18]      |                    [META] indent:
[L:  3, P: 18]      |                    newline:                                  '\n'
[L:  4, P:  1]      |                    newline:                                  '\n'
[L:  5, P:  1]      |                    whitespace:                               '    '
[L:  5, P:  5]      |                    select_statement:
[L:  5, P:  5]      |                        select_clause:
[L:  5, P:  5]      |                            keyword:                          'select'
[L:  5, P: 11]      |                            [META] indent:
[L:  5, P: 11]      |                            newline:                          '\n'
[L:  6, P:  1]      |                            whitespace:                       '        '
[L:  6, P:  9]      |                            select_clause_element:
[L:  6, P:  9]      |                                column_reference:
[L:  6, P:  9]      |                                    identifier:               'user_uuid'
[L:  6, P: 18]      |                            comma:                            ','
[L:  6, P: 19]      |                            newline:                          '\n'
[L:  7, P:  1]      |                            whitespace:                       '        '
[L:  7, P:  9]      |                            select_clause_element:
[L:  7, P:  9]      |                                column_reference:
[L:  7, P:  9]      |                                    identifier:               'email'
[L:  7, P: 14]      |                        newline:                              '\n'
[L:  8, P:  1]      |                        whitespace:                           '    '
[L:  8, P:  5]      |                        [META] dedent:
[L:  8, P:  5]      |                        from_clause:
[L:  8, P:  5]      |                            keyword:                          'from'
[L:  8, P:  9]      |                            whitespace:                       ' '
[L:  8, P: 10]      |                            from_expression:
[L:  8, P: 10]      |                                [META] indent:
[L:  8, P: 10]      |                                from_expression_element:
[L:  8, P: 10]      |                                    table_expression:
[L:  8, P: 10]      |                                        table_reference:
[L:  8, P: 10]      |                                            identifier:       '`dataset`'
[L:  8, P: 10]      |                                            dot:              '.'
[L:  8, P: 10]      |                                            identifier:       '`accounts_public`'
[L:  8, P: 10]      |                                            dot:              '.'
[L:  8, P: 10]      |                                            identifier:       '`account_data_*`'
[L:  8, P: 50]      |                                [META] dedent:
[L:  8, P: 10]      |                    newline:                                  '\n'
[L:  8, P: 10]      |                    newline:                                  '\n'
[L:  8, P: 50]      |                    [META] dedent:
[L:  8, P: 10]      |                    end_bracket:                              ')'
[L:  8, P: 10]      |            newline:                                          '\n'
[L:  8, P: 10]      |            newline:                                          '\n'
[L:  8, P: 10]      |            select_statement:
[L:  8, P: 10]      |                select_clause:
[L:  8, P: 10]      |                    keyword:                                  'select'
[L:  8, P: 50]      |                    [META] indent:
[L:  8, P: 10]      |                    whitespace:                               ' '
[L:  8, P: 10]      |                    select_clause_element:
[L:  8, P: 10]      |                        wildcard_expression:
[L:  8, P: 10]      |                            wildcard_identifier:
[L:  8, P: 10]      |                                star:                         '*'
[L:  8, P: 10]      |                whitespace:                                   ' '
[L:  8, P: 50]      |                [META] dedent:
[L:  8, P: 10]      |                from_clause:
[L:  8, P: 10]      |                    keyword:                                  'from'
[L:  8, P: 10]      |                    whitespace:                               ' '
[L:  8, P: 10]      |                    from_expression:
[L:  8, P: 10]      |                        [META] indent:
[L:  8, P: 10]      |                        from_expression_element:
[L:  8, P: 10]      |                            table_expression:
[L:  8, P: 10]      |                                table_reference:
[L:  8, P: 10]      |                                    identifier:               'account_data'
[L:  8, P: 10]      |                        newline:                              '\n'
[L:  8, P: 50]      |                        [META] dedent:
[L:  8, P: 10]      |                    unparsable:                               !! Expected: 'Nothing...'
[L:  8, P: 10]      |                        raw:                                  ')'
[L:  8, P: 10]      |                        comma:                                ','
[L:  8, P: 10]      |                        raw:                                  'account_data'
[L:  8, P: 10]      |                        whitespace:                           ' '
[L:  8, P: 10]      |                        raw:                                  'as'
[L:  8, P: 10]      |                        whitespace:                           ' '
[L:  8, P: 10]      |                        raw:                                  '('
[L:  8, P: 10]      |                        newline:                              '\n'
[L:  8, P: 10]      |                        newline:                              '\n'
[L:  8, P: 10]      |                        whitespace:                           '    '
[L:  8, P: 10]      |                        raw:                                  'select'
[L:  8, P: 10]      |                        newline:                              '\n'
[L:  8, P: 10]      |                        whitespace:                           '        '
[L:  8, P: 10]      |                        raw:                                  'user_uuid'
[L:  8, P: 10]      |                        comma:                                ','
[L:  8, P: 10]      |                        newline:                              '\n'
[L:  8, P: 10]      |                        whitespace:                           '        '
[L:  8, P: 10]      |                        raw:                                  'email'
[L:  8, P: 10]      |                        newline:                              '\n'
[L:  8, P: 10]      |                        whitespace:                           '    '
[L:  8, P: 10]      |                        raw:                                  'from'
[L:  8, P: 10]      |                        whitespace:                           ' '
[L:  8, P: 10]      |                        raw:                                  '__dbt__cte__base_accounts__account_data'
[L:  8, P: 50]      |                        newline:                              '\n'
[L:  9, P:  1]      |                        newline:                              '\n'
[L: 10, P:  1]      |                        raw:                                  ')'
[L: 10, P:  2]      |                        newline:                              '\n'
[L: 11, P:  1]      |                        newline:                              '\n'
[L: 12, P:  1]      |                        raw:                                  'select'
[L: 12, P:  7]      |                        whitespace:                           ' '
[L: 12, P:  8]      |                        raw:                                  '*'
[L: 12, P:  9]      |                        whitespace:                           ' '
[L: 12, P: 10]      |                        raw:                                  'from'
[L: 12, P: 14]      |                        whitespace:                           ' '
[L: 12, P: 15]      |                        raw:                                  'account_data'
[L: 12, P: 27]      |    newline:                                                  '\n'

==== parsing violations ====
L:   8 | P:  10 |  PRS | Line 13, Position 1: Found unparsable section: '),account_data as
                       | (\n\n select\n ...'
```

**Comment by barrywhart** [comment]
Created #1788 to address #1437 (and this issue). Hoping it'll be tested, reviewed, and merged in the next 1-2 weeks.

**Comment by barrywhart** [comment]
This was addressed in #1788. Closing this issue.

BTW, 👍🏽 for using the word `pants` in an example. Fun!

👖 forever!

**Comment by jdub** [comment]
Hooray, thanks! 👖
